### PR TITLE
feature/provider validation

### PR DESCRIPTION
**Pull request by Lucaas-Eduardo**
Added adm login which when logging redirects to the adm control panel.

In the control panel adm has the ability to accept or reject new provider accounts (accounts with the variables: adm, provider and client set to false), if the account is rejected, the account is deleted
and if accepted, the provider value is changed to true.

Trello: https://trello.com/c/uS639HHF

### CODE REVIEW

File: app/views/adms/control_panel/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<p id="notice"><%= notice %></p>
+
+<h1 style="text-align: center">Painel do ADM</h1>
```
Comment (renatamarques97) [inline app/views/adms/control_panel/index.html.erb:3]: please add I18n here

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -31,7 +29,9 @@
         <%= link_to "Login Client",     new_client_session_path,        :class => 'navbar-link'  %>
         <%= link_to "Sign up Provider", new_provider_registration_path, :class => 'navbar-link'  %> |
         <%= link_to "Login Provider",   new_provider_session_path,      :class => 'navbar-link'  %>
+        <%= link_to "Login ADM",        new_adm_session_path,           :class => 'navbar-link'  %>
```
Comment (renatamarques97) [inline app/views/layouts/application.html.erb:32]: add i18n here

File: app/models/adm.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
-require "singleton"
+#require "singleton"
 
 class Adm < Person
-  include Singleton
+  #include Singleton
```
Comment (renatamarques97) [inline app/models/adm.rb:4]: why did you remove singleton?

File: app/controllers/adms/control_panel_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  def reject_provider
+    provider_id = params[:provider_id]
+
+    provider = Person.find(provider_id)
+
+    provider.destroy
+    redirect_to "/adms/control_panel"
+  end
+
+
```
Comment (renatamarques97) [inline app/controllers/adms/control_panel_controller.rb:24]: remove these lines

File: app/controllers/adms/sessions_controller.rb [hunk 5]
```diff
@@ -1,4 +1,12 @@
 # frozen_string_literal: true
 
-class Adm::SessionsController < Devise::SessionsController
+class Adms::SessionsController < Devise::SessionsController
+  def create
```
Comment (renatamarques97) [inline app/controllers/adms/sessions_controller.rb:4]: this is the super method?

File: app/controllers/adms/control_panel_controller.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+class Adms::ControlPanelController < ApplicationController
+  def index
```
Comment (renatamarques97) [inline app/controllers/adms/control_panel_controller.rb:2]: where is the spec for this controller?

File: app/controllers/adms/sessions_controller.rb [hunk 7]
```diff
@@ -1,4 +1,12 @@
...
+    self.resource = warden.authenticate!(auth_options)
+    set_flash_message!(:notice, :signed_in)
+    sign_in(resource_name, resource)
+    yield resource if block_given?
+    redirect_to "/adms/control_panel"
+  end
+
```
Comment (renatamarques97) [inline app/controllers/adms/sessions_controller.rb:11]: remove this line

File: app/controllers/adms/sessions_controller.rb [hunk 8]
```diff
@@ -1,4 +1,12 @@
 # frozen_string_literal: true
 
-class Adm::SessionsController < Devise::SessionsController
+class Adms::SessionsController < Devise::SessionsController
+  def create
+    self.resource = warden.authenticate!(auth_options)
+    set_flash_message!(:notice, :signed_in)
+    sign_in(resource_name, resource)
+    yield resource if block_given?
+    redirect_to "/adms/control_panel"
```
Comment (renatamarques97) [inline app/controllers/adms/sessions_controller.rb:9]: please check the redirects to not conflict with others.

**Review by renatamarques97 — CHANGES_REQUESTED** [review]

File: app/controllers/adms/control_panel_controller.rb [hunk 9]
```diff
@@ -1,25 +1,24 @@
...
     @people = Person.where(admin: false, provider: false, client:false)
   end
 
   def accept_provider
-    provider_id = params[:provider_id]
-
-    provider = Person.find(provider_id)
-
-    provider.update(provider: true)
+    @_provider.update(provider: true)
```
Comment (renatamarques97) [inline app/controllers/adms/control_panel_controller.rb:9]: much better this way

File: app/models/adm.rb (hunk 3 above)
Comment (Lucaas-Eduardo) [reply app/models/adm.rb:4]: was commented so that the class can be instantiated normally, I will delete

**Review by AleBL — APPROVED** [review]

**Review by renatamarques97 — APPROVED** [review]

**Comment by AleBL** [comment]
[Painel do ADM](https://trello.com/c/uS639HHF/33-painel-do-adm)

**Comment by AleBL** [comment]
[cadastro de fornecedor deve ser aprovado pelo adm](https://trello.com/c/NcI904sF/26-cadastro-de-fornecedor-deve-ser-aprovado-pelo-adm)
